Added check for duplicate dataframe columns

philippjfr opened the pull request:
- [x] Addresses https://github.com/ioam/holoviews/issues/2476
- [x] Adds unit test

philippjfr commented:
Ready to merge.

jbednar commented on holoviews/core/data/pandas.py:
```diff
@@ -80,6 +81,16 @@ def init(cls, eltype, data, kdims, vdims):
+                count = col_counter[d]
+                if d in col_counter and count > 1:
+                    raise DataError('Dimensions may not reference duplicated DataFrame '
+                                    'columns, found %d %r columns. If you want to plot '
```
Maybe `columns (found %d %r columns).`

jlstevens commented on holoviews/core/data/pandas.py:
```diff
@@ -80,6 +81,16 @@ def init(cls, eltype, data, kdims, vdims):
+                    raise DataError('Dimensions may not reference duplicated DataFrame '
+                                    'columns (found %d %r columns). If you want to plot '
+                                    'a column against itself simply declare two dimensions '
+                                    'with the same name. '% (count, d), cls)
```
I don't really see a need to use ``Counter`` for this. I think I would have done something like:

```python
columns = []
for d in kdims+vdims:
   d = d.name if isinstance(d, Dimension) else d
    if d in columns:
        raise DataError('Dimensions may not reference duplicated DataFrame '
                                     'column  %r. If you want to plot '
                                    'a column against itself simply declare two dimensions '
                                    'with the same name. '% d, cls)
   columns.append(d)
```

It is just as long as avoids an import.

philippjfr replied:
That logic is wrong, we're checking for referenced columns that are duplicated in the dataframe, not duplicate dimensions. Happy to rewrite it without ``Counter`` though.

philippjfr commented:
Some issues with test data, seems some bad data was captured or merged at some point causing a bokeh elements test to fail, going to rebuild and merge this to keep other stuff moving. I've addressed all the comments and it's not a controversial PR.
